src/app/shared/plnkr/plnkr-util.ts

```diff
@@ -0,0 +1,109 @@
+import { Http } from '@angular/http';
+
+export class PlunkerUtil {
+  PLUNKER_URL: string = 'http://plnkr.co/edit/?p=preview';
+
+  COPYRIGHT =
+    `Copyright 2016 Google Inc. All Rights Reserved.
+    Use of this source code is governed by an MIT-style license that
+    can be found in the LICENSE file at http://angular.io/license`;
+
+  TEMPLATE_PATH = '/app/shared/plnkr/template/';
+  TEMPLATE_FILES = ['index.html', 'systemjs.config.js', 'main.ts'];
+
+  TAGS: string[] = ['angular', 'material', 'example'];
+
+  description: string = 'Example for Angular Material';
+  examplePath: string = '/app/shared/plnkr/example/';
+  exampleFiles: string[] = ['button-demo.html', 'button-demo.scss', 'button-demo.ts'];
+  selectorName: string = 'button-demo';
+  indexFilename: string = 'button-demo';
+  componentName: string = 'ButtonDemo';
+
+  form: HTMLFormElement;
+
+  constructor(private _http: Http) {}
+
+  /** Construct the plunker content */
+  initPlunker(examplePath: string,
+              exampleFiles: string[],
+              selectorName: string,
+              indexFilename: string,
+              componentName: string,
+              description: string) {
+    this.examplePath = examplePath;
+    this.exampleFiles = exampleFiles;
+    this.selectorName = selectorName;
+    this.indexFilename = indexFilename;
+    this.componentName = componentName;
+    this.description = description;
+
+    this.form = this._setupForm();
+
+    for (let file of this.TEMPLATE_FILES) {
+      this._readFile(file, this.TEMPLATE_PATH);
+    }
+
+    for (let file of this.exampleFiles) {
+      this._readFile(file, this.examplePath);
+    }
+
+    for (let i = 0; i < this.TAGS.length; i++) {
+      this._generateInput(`tags[${i}]`, this.TAGS[i]);
+    }
+
+    this._generateInput('private', 'true');
+    this._generateInput('description', this.description);
+  }
+
+  /** Open plunker link */
+  openPlunker(): void {
+    this.form.submit();
+  }
+
+  _setupForm(): HTMLFormElement {
+    var form = document.createElement('form');
+    form.action = this.PLUNKER_URL;
+    form.method = 'post';
+    form.target = '_blank';
+    return form;
+  }
+
+  _generateInput(name: string, value: string, isFile = false) {
+    var input = document.createElement('input');
+    input.type = 'hidden';
+    input.name = (isFile ? `files[${name}]` : name);
+    input.value = value;
+    this.form.appendChild(input);
+  }
+
+  _readFile(filename: string, path: string) {
+    this._http.get(path + filename).subscribe(
+      (response) => {
+        let content = response.text();
+        content = this._processContent(filename, path, content);
+        this._generateInput(filename, content, true);
```

Comment:
I'd break these out into separate handler functions:
```ts
 this._http.get(path + filename).subscribe(
  response => this._addFileToForm(response.text()),
  error => console.log(error));
```

Where `_addFileToForm` massages the file content and adds the input. Then you could remove the `isFile` argument from `_createFormInput` and pass the `files[${name}]` as the name (since you know specifically you're processing a file).